### ISSUE TITLE
Fix S4056: Rule should not recommend to use obsolete methods

### ISSUE THREAD

**Issue by gkello** [SonarSource/sonar-csharp#708]
### Description

The usage of Enum.ToString() method is detected by S4056 rule, but the overloads which are using IFormatProvider are depreciated, as described in https://docs.microsoft.com/en-us/dotnet/api/system.enum.tostring?view=netframework-4.7

> ToString(String, IFormatProvider)	This method overload is obsolete; use ToString(String).
> ToString(IFormatProvider)	This method overload is obsolete; use ToString().

###

 Repro steps

```C#
string str1 = DayOfWeek.Monday.ToString();
string str2 = DayOfWeek.Monday.ToString("F");
```
### Expected behavior

No issues for S4056 is detected

### Actual behavior

Both lines are generating 
>Warning	S4056	Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.	

### Related information

* SonarAnalyzer.CSharp 6.3.0.2862
* Visual Studio Professional 2017, version 15.3.1

**Comment by Evangelink** [comment]
Hi @gkello,

Thanks for the feedback. You are right this usage is deprecated since fwk 2.0, we will fix this soon.

- Update [RSPEC-4056](https://jira.sonarsource.com/browse/RSPEC-4056) to include exception
- Update rule to conform to new behavior.
